Flaky test TestServiceWithPredefinedNetwork

This test was added recently in https://github.com/moby/moby/pull/36316, but is failing on a lot of PR's with:

```
13:46:02 --- FAIL: TestServiceWithPredefinedNetwork (12.36s)
13:46:02 	daemon.go:283: [d47ab7ceb5568] waiting for daemon to start
13:46:02 	daemon.go:315: [d47ab7ceb5568] daemon started
13:46:02 	service_test.go:52: timeout hit after 10s: task count at 1 waiting for 0
13:46:02 	daemon.go:273: [d47ab7ceb5568] exiting daemon
13:46:02 FAIL
```

Failure is in the teardown of the test; https://github.com/moby/moby/blob/0b0af855ae28698f8e3dc97cdc2a46e714ce6b6a/integration/network/service_test.go#L52


/cc @selansen

--- a/integration/network/service_test.go
+++ b/integration/network/service_test.go
@@ -6,7 +6,6 @@ import (
 	"time"
 
 	"github.com/docker/docker/api/types"
-	"github.com/docker/docker/api/types/filters"
 	swarmtypes "github.com/docker/docker/api/types/swarm"
 	"github.com/docker/docker/client"
 	"github.com/docker/docker/integration/internal/swarm"
@@ -51,10 +50,6 @@ func TestServiceWithPredefinedNetwork(t *testing.T) {
 
 	err = client.ServiceRemove(context.Background(), serviceID)
 	assert.NilError(t, err)
-
-	poll.WaitOn(t, serviceIsRemoved(client, serviceID), pollSettings)
-	poll.WaitOn(t, noTasks(client), pollSettings)
-
 }
 
 const ingressNet = "ingress"
@@ -108,7 +103,7 @@ func TestServiceWithIngressNetwork(t *testing.T) {
 	assert.NilError(t, err)
 
 	poll.WaitOn(t, serviceIsRemoved(client, serviceID), pollSettings)
-	poll.WaitOn(t, noTasks(client), pollSettings)
+	poll.WaitOn(t, noServices(client), pollSettings)
 
 	// Ensure that "ingress" is not removed or corrupted
 	time.Sleep(10 * time.Second)
@@ -125,8 +120,6 @@ func TestServiceWithIngressNetwork(t *testing.T) {
 
 func serviceRunningCount(client client.ServiceAPIClient, serviceID string, instances uint64) func(log poll.LogT) poll.Result {
 	return func(log poll.LogT) poll.Result {
-		filter := filters.NewArgs()
-		filter.Add("service", serviceID)
 		services, err := client.ServiceList(context.Background(), types.ServiceListOptions{})
 		if err != nil {
 			return poll.Error(err)
@@ -158,5 +151,19 @@ func swarmIngressReady(client client.NetworkAPIClient) func(log poll.LogT) poll.
 			return poll.Continue("ingress not ready: does not contain the ingress-sbox")
 		}
 		return poll.Success()
+	}
+}
+
+func noServices(client client.ServiceAPIClient) func(log poll.LogT) poll.Result {
+	return func(log poll.LogT) poll.Result {
+		services, err := client.ServiceList(context.Background(), types.ServiceListOptions{})
+		switch {
+		case err != nil:
+			return poll.Error(err)
+		case len(services) == 0:
+			return poll.Success()
+		default:
+			return poll.Continue("Service count at %d waiting for 0", len(services))
+		}
 	}
 }

